src/main/scala/org/apache/spark/sql/delta/DeltaStatsCollectionUtils.scala

```diff
@@ -19,50 +22,115 @@ import org.apache.spark.sql.SparkSession
 
 trait DeltaStatsCollectionUtils {
 
+  /**
+   * Return a tuple of (outputStatsCollectionSchema, statsCollectionSchema).
+   * outputStatsCollectionSchema is the data source schema from DataFrame used for stats
+   * collection. It contains the columns in the DataFrame output, excluding the partition columns.
+   * tableStatsCollectionSchema is the schema to collect stats for. It contains the columns in the
+   * table schema, excluding the partition columns. Note: We only collect NULL_COUNT stats (as the
+   * number of rows) for the columns in statsCollectionSchema but missing in
+   * outputStatsCollectionSchema
+   */
+  protected def getStatsSchema(
+      snapshot: Snapshot,
+      outputSchema: StructType,
+      dataFrameOutput: Seq[Attribute],
+      partitionSchema: StructType): (Seq[Attribute], Seq[Attribute]) = {
+
+    // Column mapping mode
+    val columnMappingMode = snapshot.columnMappingMode
+    // Stats Schema: if the table is not initialized, we use the output schema to collect stats;
+    // otherwise, we use the table schema to collect stats
+    val statsSchema =
```

Comment:
The failing test was not finding the Delta File Stats on the first write. 

```scala

      val data = Seq(("A", 1), ("B", 2), ("C", 3)).toDF("a", "b")
      data.write
        .format("qbeast")
        .option("columnsToIndex", "a,b")
        .save(tmpDir)

      val stats =
        DeltaLog.forTable(spark, tmpDir).unsafeVolatileSnapshot.allFiles.collect().map(_.stats)
      stats.length shouldBe >(0)
      stats.head shouldBe "{\"numRecords\":3,\"minValues\":{\"a\":\"A\",\"b\":1}," +
        "\"maxValues\":{\"a\":\"C\",\"b\":3}," +
        "\"nullCount\":{\"a\":0,\"b\":0}}"

```

The problem was on `DeltaStatsCollectionUtils`, which **was initializing the `StatsCollector` from an empty schema** (since the Metadata of the Snapshot was empty because it was the first write of a Table). 

Hope this small fix can solve the issue.